gasmodel/Pact/Core/GasModel/ModuleDepsBench.hs

```diff
@@ -0,0 +1,328 @@
+
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE DataKinds #-}
+
+module Pact.Core.GasModel.ModuleDepsBench where
+
+-- |
+-- Module      :  Pact.Core.GasModel.ModuleDepsBench
+-- Copyright   :  (C) 2025 Kadena
+-- License     :  BSD-style (see the file LICENSE)
+-- Maintainer  :  Jose Cardona <jose@kadena.io>
+--
+-- Benchmarks for our transitive dependency handling
+--
+import Criterion as C
+import Control.Monad.State
+import Control.Monad
+import Control.Monad.Reader
+import Control.Lens
+import Data.IORef
+import Pact.Core.Names
+import Pact.Core.Hash
+import Pact.Core.Gen
+import Pact.Core.IR.Term
+
+import Hedgehog hiding (Var)
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import qualified Data.Set as S
+import NeatInterpolation (text)
+import Pact.Core.Builtin
+import Pact.Core.Info (LineInfo)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Pact.Core.Type
+import Data.Default
+import Pact.Core.Capabilities
+import Pact.Core.Guards
+import Pact.Core.Environment
+import Pact.Core.TransitiveDependencies
+import Pact.Core.Persistence
+import Pact.Core.PactValue
+import Pact.Core.Errors
+import Pact.Core.SPV
+import Pact.Core.Namespace
+import Pact.Core.Gas
+import Pact.Core.Persistence.MockPersistence (mockPactDb)
+import Pact.Core.Serialise (serialisePact_lineinfo_pact51)
+import Pact.Core.GasModel.ModuleLoadBench(sampleWithSeed)
+
+data DPGenState
+  = DPGenState
+  { _validIdents :: M.Map FullyQualifiedName (S.Set FullyQualifiedName)
+  -- ^ A map from a generated fully qualified name to the set of its dependencies
+  , _currentDef :: FullyQualifiedName
+  -- ^ The current definition for which we are generating terms
+  , _existingModules :: S.Set ModuleName
+  -- ^ The set of existing modules
+  , _allDefs :: M.Map FullyQualifiedName (EvalDef CoreBuiltin LineInfo)
+  -- ^ The map of all generated defs
+  }
+
+makeLenses 'DPGenState
+
+-- | A DepGenM action:
+--   - Can generate a term within a particular module context
+--   - Can record the set of dependencies for a newly generated function
+type DepGenM a = ReaderT (ModuleName, ModuleHash) (StateT DPGenState Gen) a
+
+
+emptyModule :: ModuleName
+emptyModule = ModuleName "" Nothing
+emptyModuleHash :: ModuleHash
+emptyModuleHash = ModuleHash (Hash "")
+
+-- | Run a DepGenM action and return the set of all generated definitions
+runDepGenM :: DepGenM a -> Gen (a, M.Map FullyQualifiedName (EvalDef CoreBuiltin LineInfo))
```

Comment:
Most of our `DepGenM` code is copy pasted from `TransitiveDependencyTests.hs` but with minor tweaks to intentionally generate larger modules for benchmarks.